### ISSUE TITLE
Update-check is failing on Windows

### ISSUE THREAD

**Issue by LSmyrnaios** [meganz/MEGAsync#350]
Hi,
I would like to report that MEGAsync on Windows reports that no available update exists, even though I am having v.4.2.3, while the v.4.2.4 is out.
This problem exist for any new version.
Each time I have to wait weeks before the Desktop app realizes that a new version is out.

I am running Windows 10 v.1903

**Comment by polmr** [comment]
Please, @LSmyrnaios , try again today.

**Comment by LSmyrnaios** [comment]
@polmr I tried, but with no luck.
I checked manually multiple times, I left it open for a period of time, in order to be able to automatically get the update itself (as it usually does, even while reporting "no new updates are available") and I restarted it.

**Comment by polmr** [comment]
Autoupdate is available for version 4.2.5. Sooner or later should be offered to you. If the problem persist, please, try downloading the installer manually from https://mega.nz/sync. Reinstalling will keep your preferences and session.

**Comment by LSmyrnaios** [comment]
I installed manually the v.4.2.5
I will report about the next release that will come out, weather I can get it as an update or not,

**Comment by Perkolator** [comment]
Is the update check working on Windows at all? The update check from the settings always says "No update available at this time"!

I complained about this same problem [in here](https://github.com/meganz/MEGAsync/issues/330#issuecomment-520254801) last fall, and now update "v4.3.0.8_Win" has been out for 8 days and the update check still doesn't pick it up.

I haven't seen a single update found by the client. Something is obviously wrong. Please investigate and fix this. Thank you.

**Comment by LSmyrnaios** [comment]
I confirm, MEGASync still has this bug, it reports "No update available at this time".
I have checked the option to "Update automatically", and also clicked the "Check for updates" button.

**Comment by Perkolator** [comment]
12 days since "v4.3.0.8_Win" release and still no update found by the client. I really hope somebody would fix this, it's not working.

**Comment by Perkolator** [comment]
20 days since the last release and finally today when checking the settings (I manually checked for updates every day, automatic update is off), the "check for updates" button had turned to "install" button.

20 days is too long, especially if there's some security updates. Why does it take so long for the client to pick up new versions? Especially when user manually runs the update check.

**Comment by LSmyrnaios** [comment]
Hi, I am still getting "No update available at this time"..
I will just go and install the new version.. and report again when the next release comes out.

**Comment by Perkolator** [comment]
And again same problem. I just don't understand what's the problem and why the staff here pretty much ignores this (and a lot of other issues too). It's now 21 days since "v4.3.3.0_Win" was released and my Windows client (currently "v4.3.1.0_Win") still doesn't detect the new version. This is ridiculous.

**Comment by mattw-mega** [comment]
If you are very keen to get the latest update, you can download it from the website - note that the installer is for 64 bit MEGAsync now, and if you want to continue with 32 bit, there's an extra link to get that one.  cheers

**Comment by Perkolator** [comment]
But this open issue is about the very slow client update detection, not about how one could download and install the client manually. Why does it take so long for the client to detect a new version? Why is the staff here so ignorant to our reports? Is this not a problem according to the staff? Maybe you should then say it and close this issue? 3 weeks (and still counting) is a ridiculous wait for the client to pick up a new update.

**Comment by LSmyrnaios** [comment]
I confirm, MEGASync still has this bug.
I cannot understand why this issue (and other important ones) has not been resolved yet..

**Comment by Perkolator** [comment]
Just now, I did a manual check again and the client said "Update available. Downloading". But nothing happened and the popup(s) went away. Tried the check update button again and now the popup says "no update available at this time". And 3rd try with the check update button, again "Update available. Downloading" and then I was offered to update the client. Odd that the check updates function doesn't seem to work properly when an update is actually available. Odd coincidence, this time update is quickly available after writing here.

If the about 3 weeks waiting period is normal, then say that is the case here and close this issue. If it's not normal, try to fix it. Thank you very much.

**Comment by LSmyrnaios** [comment]
@Perkolator nothing in what you described can be "normal", nor the waiting period, nor the procedure.
I think this issue should stay open until the update function is fixed.

I am still getting "No update available at this time".

**Comment by Perkolator** [comment]
My opinion is the same as yours, I just tried to get the staff to react to this in any way. The worst thing is the lack of communication from the staff.

**Comment by Perkolator** [comment]
And again. 4.3.4.0 was released 20 days ago, and newer 4.3.5.0 was released 14 days ago. My Windows client with 4.3.3 version says "no update available at this time". You devs don't seem to care at all..

**Comment by LSmyrnaios** [comment]
I get the same message..

**Comment by vtmateos** [comment]
Hi @Perkolator  and @LSmyrnaios ,

Apologies for the inconveniences. 4.3.5 is on staged rollout and you will receive the update in the upcoming days. If you do not want to wait, you can make a fresh install from our main webpage and download 4.3.5.

Thanks.

**Comment by Perkolator** [comment]
Yes, thank you, but why does it always take so long to receive the updates through the client?

**Comment by Perkolator** [comment]
Today when manually checking, the 4.3.5.0 version was found and installed. 23 days waiting period, 29 days if counting from 4.3.4.0 release date.

Could somebody from the MEGA team kindly answer whether the problem described in this issue, the very long waiting period of updates reaching the client update mechanism, **is this issue a bug or not**?? Thank you! Why are you completely avoiding answering this question?!

**Comment by mattw-mega** [comment]
Hi @Perkolator ,  we are cautious with the rollout because although we test carefully, sometimes there are combinations of factors in the real world with so many users and setups, that a bug could be discovered, and we can fix that before rolling out to more users.   It sounds like you are very keen to get the update sooner, and that's great, you have the option to do that by downloading the new installer from the website.   cheers

**Comment by Perkolator** [comment]
Thank you for giving more information.

I have few suggestions/wishes:

1) If an update includes security fixes, the rollout period is far far too long, these fixes should be pushed as fast as you can.

2) With non-security updates, the ~3 weeks rollout period is still far far too long. I can't remember any other software vendor who would delay their releases so long.

3) Maybe you should try to publish beta test releases instead of doing that with stable releases?

4) You really need to work on your communication. This issue was opened on Sep 15, 2019. Over a year ago. And now, finally, somebody spoke out, after I repeatedly pleaded you to say something! It's now very clear that this issue is not a bug but intended behavior. How much time has been wasted on this issue?

**Comment by Perkolator** [comment]
Related issue: just now on my linux mint 20 laptop, update manager shows a new 4.3.5-9.1 update ready to be downloaded/installed. On github releases page, latest linux version of megasync is 4.3.3.0. What's with that? Updates are a bit messy, releases not listed on releases page, updates reach clients very late, changelogs are still somewhat poor (identical changelogs for different versions, and platforms). There's room for improvement on many levels.

**Comment by Perkolator** [comment]
I just noticed (with [SUMo](https://www.kcsoftwares.com/?sumo)) that version 4.3.9.0 (at least for windows) has been out for about a week now, the mega client of course knows nothing about it yet, as usual, but even here in github that version hasn't been published. You really need to improve the way you are handling/publishing updates! BTW. the linux 4.3.8.0 release entry is missing too.

**Comment by Perkolator** [comment]
There's definitely something wrong with the client update check. On Windows I have 4.4.0 installed and it had detected an update itself automatically (no notifications though!), selecting to update I was greeted with "There was a problem installing the update. Please try again later..." notification. I tried several times, always error. Then I closed the client, restarted it (still version 4.4.0) and ran "check for updates" from settings = "No update available at this time" notification pops up. You definitely have to think about making the UX better with the client update feature.

**Comment by Perkolator** [comment]
Again more confusion; downloaded the sync client and started installation, it says "MEGAsync 4.5.2" while the github releases page shows the "4.5.1.0" as the latest Windows version. Come on guys, you REALLY need to step up your game with the update stuff.

**Comment by bingoxo** [comment]
i'm still experiencing this

**Comment by Wrizz** [comment]
I am also not getting any auto updates AND when I update the desktop app manually, it just degrades back to 4.6.5 automatically.

**Comment by rl-mega** [comment]
Hi! As explained we do a progressive rollout.
We strive to improve release frequency, and decreased rollout time as well as code source publication in github.
Thanks for your patience.
